tests/test_drvsupport.py

```diff
@@ -10,94 +10,104 @@
 
 
 @requires_gdal24
-@pytest.mark.parametrize('format', ['GeoJSON', 'ESRIJSON', 'TopoJSON', 'GeoJSONSeq'])
+@pytest.mark.parametrize("format", ["GeoJSON", "ESRIJSON", "TopoJSON", "GeoJSONSeq"])
 def test_geojsonseq(format):
     """Format is available"""
     assert format in fiona.drvsupport.supported_drivers.keys()
 
 
-@pytest.mark.parametrize('driver', [driver for driver, raw in supported_drivers.items() if 'w' in raw])
+@pytest.mark.parametrize(
+    "driver", [driver for driver, raw in supported_drivers.items() if "w" in raw]
+)
 def test_write_or_driver_error(tmpdir, driver, testdata_generator):
     """
-        Test if write mode works.
+    Test if write mode works.
 
     """
 
     if driver == "BNA" and GDALVersion.runtime() < GDALVersion(2, 0):
         pytest.skip("BNA driver segfaults with gdal 1.11")
 
-    schema, crs, records1, _, test_equal, create_kwargs = testdata_generator(driver, range(0, 10), [])
+    schema, crs, records1, _, test_equal, create_kwargs = testdata_generator(
+        driver, range(0, 10), []
+    )
     path = str(tmpdir.join(get_temp_filename(driver)))
 
-    if (driver in driver_mode_mingdal['w'] and
-            get_gdal_version_num() < calc_gdal_version_num(*driver_mode_mingdal['w'][driver])):
+    if driver in driver_mode_mingdal[
+        "w"
+    ] and get_gdal_version_num() < calc_gdal_version_num(
+        *driver_mode_mingdal["w"][driver]
+    ):
 
         # Test if DriverError is raised for gdal < driver_mode_mingdal
         with pytest.raises(DriverError):
-            with fiona.open(path, 'w',
-                            driver=driver,
-                            crs=crs,
-                            schema=schema,
-                            **create_kwargs) as c:
+            with fiona.open(
+                path, "w", driver=driver, crs=crs, schema=schema, **create_kwargs
+            ) as c:
                 c.writerecords(records1)
 
     else:
         # Test if we can write
-        with fiona.open(path, 'w',
-                        driver=driver,
-                        crs=crs,
-                        schema=schema,
-                        **create_kwargs) as c:
+        with fiona.open(
+            path, "w", driver=driver, crs=crs, schema=schema, **create_kwargs
+        ) as c:
 
             c.writerecords(records1)
 
-        if driver in {'FileGDB', 'OpenFileGDB'}:
+        if driver in {"FileGDB", "OpenFileGDB"}:
             open_driver = driver
         else:
             open_driver = None
-        with fiona.open(path, driver=open_driver) as c:
-            assert c.driver == driver
-            items = list(c)
-            assert len(items) == len(records1)
-            for val_in, val_out in zip(records1, items):
-                assert test_equal(driver, val_in, val_out), "in: {val_in}, out: {val_out}".format(val_in=val_in,
-                                                                                                  val_out=val_out)
```

Comment:
#935 and #1031 report test failures involving coordinate comparison. So I'm removing the feature comparisons. We are really only concerned about the counting of features here. The exact values can be driver/platform dependent.